dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/internal/DatabaseStoreImpl.java

```diff
@@ -699,18 +703,23 @@ protected String createOrmFileContentsForBatch(String[] entityClassNames, String
      */
     protected InMemoryMappingFile createOrmFile(String schemaName,
                                                 String tablePrefix,
+                                                LinkedHashSet<String> tableNames, // entries correspond to entityClassNames
                                                 String[] entityClassNames,
                                                 String[] entityClassEntries)
                     throws UnsupportedEncodingException {
         return ((entityClassNames == null || entityClassNames.length == 0) && (entityClassEntries == null || entityClassEntries.length == 0))
                         ? null
-                        : new InMemoryMappingFile(createOrm(schemaName, tablePrefix, entityClassNames, entityClassEntries).getBytes("UTF-8"));
+                        : new InMemoryMappingFile(createOrm(schemaName, tablePrefix, tableNames, entityClassNames, entityClassEntries).getBytes("UTF-8"));
     }
 
     /**
      * @return a generic ORM xml file, in string form, using the given schema, tablePrefix, and entity classes.
      */
-    protected String createOrm(String schemaName, String tablePrefix, String[] entityClassNames, String[] entityClassEntries) {
+    protected String createOrm(String schemaName,
+                               String tablePrefix,
+                               Set<String> tableNames, // entries correspond to entityClassNames
```

Comment:
This will only work with a LInkedHashSet 
```suggestion
                               LinkedHashSet<String> tableNames, // entries correspond to entityClassNames
```